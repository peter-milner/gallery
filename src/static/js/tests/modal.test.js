import React from 'react'
import { shallow } from 'enzyme'

import Modal from '../components/modal'

const defaultProps = {
  photo: {
    url: 'test.url',
    name: 'test',
    artist: 'test-artist'
  },
  closeModal: jest.fn()
}

const renderModal = (props) => {
  const newProps = Object.assign({}, defaultProps, props)
  return shallow(<Modal {...newProps} />)
}

describe('<Modal />', () => {
  test('Snapshot', () => {
    const renderedModal = renderModal()
    expect(renderedModal).toMatchSnapshot()
  })

  test('return null if photo is null', () => {
    let renderedModal = renderModal()
    expect(renderedModal.isEmptyRender()).toEqual(false)

    renderedModal = renderModal({ photo: null })
    expect(renderedModal.isEmptyRender()).toEqual(true)
  })

  test('clicking x will call closeModal', () => {
    const props = { closeModal: jest.fn() }
    const spy = jest.spyOn(props, 'closeModal')
    const renderedModal = renderModal(props)

    renderedModal.find('.modal-close').simulate('click')
    expect(spy).toHaveBeenCalled()
  })
})
