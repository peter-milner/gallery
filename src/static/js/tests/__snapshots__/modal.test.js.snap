// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> Snapshot 1`] = `ShallowWrapper {}`;
